document-service/doc-api/tests/unit/models/test_document.py

```diff
@@ -0,0 +1,188 @@
+# Copyright © 2019 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests to assure the document Model.
+
+Test-Suite to ensure that the document Model is working as expected.
+"""
+import copy
+
+import pytest
+
+from doc_api.models import Document, utils as model_utils
+from doc_api.models.type_tables import DocumentTypes, DocumentClasses, DocumentType
+
+
+DOC1 = {
+    'consumerDocumentId': 'T0000001',
+    'consumerFilename': 'test.pdf',
+    'consumerIdentifier': 'T0000002',
+    'documentType': 'PPR_MISC',
+    'consumerFilingDateTime': '2024-07-01T19:00:00+00:00',
+    'consumerScanDateTime': '2024-05-01T19:00:00+00:00'
+}
+TEST_DOCUMENT = Document(id=1,
+                         document_service_id='1',
+                         document_type=DocumentTypes.PPR_MISC.value,
+                         add_ts=model_utils.now_ts(),
+                         consumer_document_id='T0000001',
+                         consumer_identifier='T0000002',
+                         consumer_filename='test.pdf',
+                         consumer_filing_date=model_utils.ts_from_iso_date_noon('2024-07-01'),
+                         scan_date=model_utils.ts_from_iso_date_noon('2024-05-01'))
+
+# testdata pattern is ({id}, {has_results}, {doc_type), {doc_class})
+TEST_ID_DATA = [
+    (200000001, True, DocumentTypes.PPR_MISC.value, DocumentClasses.PPR.value),
+    (300000000, False, DocumentTypes.PPR_MISC.value, DocumentClasses.PPR.value)
+]
+TEST_DOC_SERVICE_ID_DATA = [
+    ('T0000001', True, DocumentTypes.MHR_MISC.value, DocumentClasses.MHR.value),
+    ('XXXD0000', False, DocumentTypes.MHR_MISC.value, DocumentClasses.MHR.value)
+]
+TEST_DOC_ID_DATA = [
+    ('T0000001', True, DocumentTypes.CORP_MISC.value, DocumentClasses.CORP.value),
+    ('XXXD0000', False, DocumentTypes.CORP_MISC.value, DocumentClasses.CORP.value)
+]
+# testdata pattern is ({has_doc_id}, {doc_type})
+TEST_CREATE_JSON_DATA = [
+    (True, DocumentTypes.CORP_MISC.value),
+    (False, DocumentTypes.MHR_MISC.value)
+]
+
+
+@pytest.mark.parametrize('id, has_results, doc_type, doc_class', TEST_ID_DATA)
+def test_find_by_id(session, id, has_results, doc_type, doc_class):
+    """Assert that find document by primary key contains all expected elements."""
+    if not has_results:
+        document: Document = Document.find_by_id(id)
+        assert not document
+    else:
+        save_doc: Document = Document.create_from_json(DOC1, doc_type)
```

Comment:
consider leveraging test-containers, using the postgres one?
Here we're using google pubsub emulator as an example -> https://github.com/bcgov/sbc-connect-common/blob/main/python/gcp-queue/tests/integration/test_gcp_queue_testcontainer.py#L11

Reply:
Will do.